solidity/random-beacon/tasks/staking.ts

```diff
@@ -0,0 +1,110 @@
+import { task, types } from "hardhat/config"
+
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+
+task(
+  "staking",
+  "Stakes T tokens, increases authorization and registers the operator "
+)
+  .addOptionalParam("owner", "Staking owner address", undefined, types.string)
+  .addOptionalParam("provider", "Staking provider", undefined, types.string)
+  .addOptionalParam("operator", "Staking operator", undefined, types.string)
+  .addOptionalParam("amount", "Staking amount", 1_000_000, types.int)
+  .addOptionalParam(
+    "authorization",
+    "Authorization amount",
+    undefined,
+    types.int
+  )
+  .setAction(async (args, hre) => {
+    await setup(hre, args)
+  })
+
+async function setup(
+  hre: HardhatRuntimeEnvironment,
+  args: {
+    owner: string
+    provider: string
+    operator: string
+    amount: BigInteger
+    authorization: BigInteger
+  }
+) {
+  const { ethers, helpers, getNamedAccounts } = hre
+  const { owner, provider, operator, amount, authorization } = args
```

Comment:
Instead of destructuring the `owner`, `provider` etc. from args we could define the signers based on `args.*` as proposed in other comments.